packages/core/src/Cardano/types/Cip1854PublicKey.ts

```diff
@@ -0,0 +1,55 @@
+import * as BaseEncoding from '@scure/base';
+import { InvalidStringError, OpaqueString, assertIsBech32WithPrefix, assertIsHexString } from '@cardano-sdk/util';
+
+const MAX_BECH32_LENGTH_LIMIT = 1023;
+export type Cip1854PublicKey = OpaqueString<'Cip1854PublicKey'>;
+
+export const Cip1854PublicKey = (value: string): Cip1854PublicKey => {
+  try {
+    assertIsBech32WithPrefix(value, ['addr_shared_vk', 'stake_shared_vk']);
```

Comment:
Are we not going to share the bip32 key instead of payment and stake credential (extended account public key, or xpub)? I believe we discussed this on Slack a while back and I thought we agreed on this, as it's more extensible (could potentially implement multi-delegation in the future). cc @danielmain

Reply:
> I don't understand. What do you mean by the Bip32 key?

If we decide to share the bip32 keys instead of the actual keys then we dont need this utils, as those have already their own defined bech32 encoding:

`xpub1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9h0rf4`

Reply:
It's an account-level key that is used to derive payment and stake keys